oxenss/storage/database.cpp

```diff
@@ -798,6 +822,24 @@ std::vector<std::string> Database::delete_by_timestamp(
     return get_all<std::string>(st, pubkey, to_epoch_ms(timestamp), ns);
 }
 
+void Database::revoke_subkey(
+        const user_pubkey_t& pubkey, std::array<unsigned char, 32> revoke_subkey) {
+    auto get_owner = impl->prepared_st("SELECT id FROM owners WHERE pubkey = ? AND type = ?");
+    auto insert_subkey = impl->prepared_st(
+            "INSERT INTO revoked_subkeys (owner, subkey) VALUES (?, ?) ON CONFLICT DO NOTHING "
+            "RETURNING id");
+    auto ownerid = exec_and_get<int64_t>(get_owner, pubkey);
+    exec_and_get<int64_t>(insert_subkey, ownerid, (std::string)util::view_guts(revoke_subkey));
```

Comment:
We can do this in one query with a subselect:
```sql
INSERT INTO revoked_subkeys (owner, subkey)
    VALUES ((SELECT id FROM owners WHERE pubkey = ? AND type = ?), ?)
    ON CONFLICT DO NOTHING
```
We also don't care about the returned id (I don't think we need id at all) so we can drop the `RETURNING id` and just use `exec_query(insert_subkey, pubkey, blob_binder{util::view_guts(revoke_subkey)})`.  (That `blob_binder` assumes the type also gets changed from TEXT -> BLOB).